Public/Src/Engine/Cache/Serialization/JsonTree.cs

```diff
@@ -223,110 +239,21 @@ private static void BuildStringHelper(JsonNode root, IFingerprinter wr)
-        }
 
         /// <summary>
         /// Diffs two <see cref="JsonNode"/> trees.
         /// </summary>
         /// <returns>
-        /// A string representation of the diff.
+        /// A JSON string representation of the diff.
         /// </returns>
         public static string PrintTreeDiff(JsonNode rootA, JsonNode rootB)
         {
-            var changeList = DiffTrees(rootA, rootB);
-            return PrintTreeChangeList(changeList);
-        }
+            var jsonA = Serialize(rootA);
+            var jsonB = Serialize(rootB);
 
-        /// <summary>
-        /// Formats and prints the resulting <see cref="ChangeList{T}"/> from a <see cref="DiffTrees(JsonNode, JsonNode)"/>.
-        /// </summary>
-        /// <returns>
-        /// A string representation of the change list.
-        /// </returns>
-        public static string PrintTreeChangeList(ChangeList<JsonNode> changeList)
-        {
-            var treeToPrint = new PrintNode();
-            for (int i = 0; i < changeList.Count; ++i)
-            {
-                var change = changeList[i];
-                var node = change.Value;
-                var path = new LinkedList<string>();
-                // Use "it.Parent != null" instead of "it != null" to 
-                // ignore printing the root parent node that matches the opening bracket
-                // for all JSON objects
-                for (var it = node; it.Parent != null; it = it.Parent)
-                {
-                    path.AddFirst(it.Name);
-                }
-
-                var printNode = treeToPrint;
-                // Build a tree of the change list values, placing nodes based off
-                // their positions in the old or new tree
-                foreach (var pathAtom in path)
-                {
-                    if (!printNode.Children.ContainsKey(pathAtom))
-                    {
-                        printNode.Children.Add(pathAtom, new PrintNode());
-                    }
-
-                    printNode = printNode.Children[pathAtom];
-                }
-
-                printNode.ChangedNodes.Add(change);
-            }
-
-            return treeToPrint.ToString();
-        }
-
-        /// <summary>
-        /// Diffs two trees.
-        /// </summary>
-        /// <param name="rootA">
-        /// The root of the original tree.
-        /// </param>
-        /// <param name="rootB">
-        /// The root of the transformed tree.
-        /// </param>
-        /// <returns>
-        /// A <see cref="ChangeList{JsonNode}"/> containing
-        /// the leaf nodes that differ.
-        /// </returns>
-        public static ChangeList<JsonNode> DiffTrees(JsonNode rootA, JsonNode rootB)
-        {
-            var leavesA = CollectValueNodes(rootA);
-            var leavesB = CollectValueNodes(rootB);
-
-            var changeList = new ChangeList<JsonNode>(leavesA, leavesB);
+            var diff = s_jdp.Diff(JToken.Parse(jsonA), JToken.Parse(jsonB));

```

Comment:
Ideally JsonNode would be replaced with NewtonSoft.Json JToken to so that the trees can be diffed directly with converting from JsonNode->JSON string->JToken.

JsonNode -> JToken is definitely doable, it just requires more caution than this change.